xds/internal/xdsclient/e2e_test/lds_watchers_test.go

```diff
@@ -562,6 +568,114 @@ func (s) TestLDSWatch_ResourceCaching(t *testing.T) {
 	}
 }
 
+// TestLDSWatch_ExpiryTimerFiresBeforeResponse tests the case where the client
+// does not receive an LDS response for the request that it sends. We want the
+// watch callback to be invoked with an error once the watchExpiryTimer fires.
+func (s) TestLDSWatch_ExpiryTimerFiresBeforeResponse(t *testing.T) {
+	// No need to spin up a management server since we don't want the client to
+	// receive a response for the watch being registered by the test.
+
+	// Create an xDS client talking to a non-existent management server.
+	client, err := xdsclient.NewWithConfigForTesting(&bootstrap.Config{
+		XDSServer: &bootstrap.ServerConfig{
+			ServerURI:    "dummy management server address",
+			Creds:        grpc.WithTransportCredentials(insecure.NewCredentials()),
+			TransportAPI: version.TransportV3,
+			NodeProto:    &v3corepb.Node{},
+		},
+	}, defaultTestWatchExpiryTimeout, time.Duration(0))
+	if err != nil {
+		t.Fatalf("failed to create xds client: %v", err)
+	}
+	defer client.Close()
+
+	// Register a watch for a resource which is expected to fail with an error
+	// after the watch expiry timer fires.
+	updateCh := testutils.NewChannel()
+	ldsCancel := client.WatchListener(ldsName, func(u xdsresource.ListenerUpdate, err error) {
+		updateCh.Send(xdsresource.ListenerUpdateErrTuple{Update: u, Err: err})
+	})
+	defer ldsCancel()
+
+	// Wait for the watch expiry timer to fire.
+	<-time.After(defaultTestWatchExpiryTimeout)
```

Comment:
Is there another way to do this without adding a 500 ms addition to each test time to run?